[QA] QA e2e tests

mishautkin opened the pull request:
## Description

- 500+ actual tests
- Ability to run tests for Payment and Order API, depending on env var
- Reporting to Testrail
- DDEV support
- Automated env setup scripts
> See [README.md](https://github.com/mollie/WooCommerce/tree/qa/e2e-tests/tests/qa#readme) for more info

mishautkin committed:
Added `./tests/qa` dir with actual tests for v7.9.0

- 431 actual tests (Playwright, typescript)
- Reporting to Testrail
- See README.md for more info

mmaymo reviewed (changes requested):
All looks good, but could you also remove the e2e folder if we are not going to use it? Thanks!

mishautkin committed:
Removed `./tests/e2e/` folder

mishautkin committed:
Removed `parallel-surcharge` script from `package.json`

- Surcharge tests need to be divided by type to be executed in parallel

mishautkin committed:
Updates for `surcharge` tests

- Fixed `surcharge` test title
- Removed `surcharge` project from `playwright.config.ts` (Surcharge tests need to be divided by type to be executed in parallel)

mishautkin committed:
Readme update/fix

mishautkin committed:
Updated MollieSettings.Gateway

- Updated field names according to API
- Updated respective values in multiple files

mishautkin committed:
Fix of values according to updated MollieSettings.Gateway type

mishautkin committed:
Fixes in /resources

- Updated card (visa) expiration date value
- Updated products to include voucher types
- Removed unused products

mishautkin committed:
Update in MollieApi

Input data parameter doesn't need settings node any more.

mishautkin committed:
Updates in /tests/05-transaction

- Created `payment-status` folder for respective tests
- Added 12 autotests for Card with enabled Mollie components

mishautkin committed:
Fix package.json

- update playwright version
- add sequestial-transactions script

mishautkin committed:
Fix playwright.config

- fix testMatch/testIgnore
- fix reporting to testrail

mishautkin committed:
Update README.md

mishautkin committed:
Add plugins for testing, updated mollie version to 7.10.0

mishautkin committed:
Add new gateways, remove deprecated.

mishautkin committed:
Add mollieSubscription100 product.

mishautkin committed:
Fix lint issues

mishautkin committed:
Updates for Refactored API

mishautkin committed:
Updates for Payment and Order API methods.

mishautkin committed:
Merge pull request #1027 from mollie/qa/e2e-refactored-api

Qa/e2e refactored api

mishautkin committed:
Fix surcharge fees currency for Przelewy24

mishautkin committed:
Add third-party-fees plugin

mishautkin committed:
FIxes for 8.0.1

mishautkin committed:
Add missing test IDs

mishautkin committed:
Fix data-conversion, add describe to block transactions

mishautkin committed:
Update README

mishautkin committed:
Update README

mishautkin committed:
Update README

mishautkin committed:
Fix for API method in `.env`.

- Updated Readme.
- Defined API method default value.

mishautkin committed:
Updates in tests/qa/

- update Mollie to 8.0.2
- update playwright-utils to 2.6.1
- surcharge test ID fix (bancontact)
- remove feature flags

mishautkin committed:
Fix/updates for 8.0.3

- Fix payment statuses in data-conversion.ts.
- Upgrade playwright-utils to v2.7.0.
- Remove extra unused projects.

Commit:
C3333 test case was updated.
Changed the link  for documentation.

Commit:
Merge branch 'qa/e2e-tests' of https://github.com/mollie/WooCommerce into qa/e2e-tests

mishautkin committed:
Update playwright-utils to v 2.7.2 in package.json and mollie plugin .zip in resources/

mishautkin committed:
Fixes for v8.0.6

mishautkin committed:
Update mollie to v8.0.6-beta2, add test:all npm script

mishautkin committed:
Adjustment and Readme update for DDEV

mishautkin committed:
Add Mollie setup scripts

mishautkin committed:
Fix config type in multiple files

mishautkin committed:
Fix variable product

mishautkin committed:
Fix config type in woocommerce-config

mishautkin committed:
Add auto env setup scripts and update readme

mishautkin committed:
Merge branch 'develop' of https://github.com/mollie/WooCommerce into qa/e2e-tests

mishautkin committed:
Remove e2e dir

mishautkin committed:
Fix test IDs and summaries

mishautkin committed:
Fix test summaries for mollie components

mishautkin committed:
Rename type MollieTestData.PaymentStatus to MollieTestData.Transaction

mishautkin committed:
Update data-conversion.ts

mishautkin committed:
Rename process-mollie-payment-status.scenario to checkout-test-helpers

mishautkin committed:
Add reusable functions to checkout-test-helpers.ts

mishautkin committed:
Fix MollieGateway import

mishautkin committed:
Merge pull request #1108 from mollie/qa/block-refactoring

Qa/block refactoring

mishautkin committed:
Add multistep functionality

mishautkin committed:
Merge branch 'qa/e2e-tests' of https://github.com/mollie/WooCommerce into qa/germanized-multi-step-checkout

mishautkin committed:
Fix mandatory state select for Italy

mishautkin committed:
Add test C1729452, fix settings tests for assertions in block checkout

mishautkin committed:
Add test C1729452, fix settings tests for assertions in block checkout

mishautkin committed:
Merge branch 'qa/e2e-tests' of https://github.com/mollie/WooCommerce into qa/germanized-multi-step-checkout

mishautkin committed:
Add plugin Enable Bizum

mishautkin committed:
Fix screenshot on failure in playwright.config

mishautkin committed:
Merge branch 'qa/e2e-tests' of https://github.com/mollie/WooCommerce into qa/germanized-multi-step-checkout

mishautkin committed:
Add timeouts to multistep classic checkout

mishautkin committed:
Merge pull request #1110 from mollie/qa/germanized-multi-step-checkout

Qa/germanized multi step checkout

mishautkin committed:
Gitignore all playwright-report dirs

mishautkin committed:
Update playwright-utils to v2.7.4

mishautkin committed:
Fix test ID

mishautkin committed:
Enable video recording and trace for visitorPage fixture

mishautkin committed:
Enable video recording and trace for visitorPage fixture

mishautkin committed:
Fix flows for some gateways on checkouts

mishautkin committed:
Remove retry functionality from wooCommerceorderEdit.assertOrderDetails

mishautkin committed:
Fix flows for some gateways on checkouts

mishautkin committed:
Fix with timeout flows for some gateways on classic checkout

mishautkin committed:
Fix for Card container

mishautkin committed:
Update gitignore

mishautkin committed:
Ipdate playwright-utils to v2.8.0

mishautkin committed:
Update .gitignore

mishautkin committed:
Increase timeout in test C1729452

mishautkin committed:
Remove expect for Terms&conditions checkbox (Multistep)

mishautkin committed:
Fix for C419984

mishautkin committed:
Remove tests for Payconiq (gateway excluded by client)

mishautkin committed:
Fix version in mollie-plugin

mishautkin committed:
Add subscription renewal comments

mishautkin committed:
Lint fix

mishautkin committed:
Comment payconiq gateway

mishautkin committed:
Ungitignore package-lock

mishautkin committed:
Add/fix checkout locators

mishautkin committed:
Add/fix gateways

mishautkin committed:
Add Vipps, MobilePay tests

mishautkin committed:
Fix giftcard gateway

mishautkin committed:
Add `paymentOptionLabel` locator to checkout pages

mishautkin committed:
Add tests for Pay By Bank open status

mishautkin committed:
Merge branch 'develop' of https://github.com/mollie/WooCommerce into qa/e2e-tests

mishautkin committed:
Update method names

- Rename type StoreSettings to ShopConfig
- Rename utils.installAndActivateMollie method
- Add products creation to utils.configureStore method

mishautkin committed:
Move data-conversion.ts to utils/

mishautkin committed:
Refactor transactions test data

- Add MollieTestData.ShopOrder
- Remove MollieTestData.Transaction type
- Update MolliePayment type
- Remove data-concersion
- Add transaction-base-order.data
- Update all transaction test data
- Update naming in transaction scenarios

mishautkin committed:
Merge pull request #1155 from mollie/dev/qa/subscription-tests

[QA] Refactor transactions test data

mishautkin committed:
Update subscription tests

- Move subscription tests from 06- to 07-subscription
- Add data, scenario and spec for subscription-order and subscription-renewal
- Add mollieApi methods

mishautkin committed:
Update gitignore

mishautkin committed:
Move transaction tests from payment-status to 05-transaction dir

mishautkin committed:
Add assertions for order notes to transaction tests

mishautkin committed:
Move triggerSubscriptionRenewal and getSubscriptionRenewalOrderIds from mollie-api to playwright-utils

mishautkin committed:
Add ShopRefund type

mishautkin committed:
Add refund tests

mishautkin committed:
Remove deleteAllSubscriptions from beforeAll

mishautkin committed:
Fix subscriptionRenewal data

mishautkin committed:
Fix restoreCustomer method

mishautkin committed:
Add mollieClientApi fixture, fix customerSubscriptions fixture

mishautkin committed:
Fix refund methods on Order Edit page

mishautkin committed:
Fix subscription-order scenario

mishautkin committed:
Fix subscription-renewal scenario

mishautkin committed:
Update and rename mollieApi.setApiMethod to mollieApi.setAdvancedSettings

mishautkin committed:
Update playwright-utils to v3.0.0

mishautkin committed:
Lint fix and expect error messages updates

mishautkin committed:
Add order-notes-helpers

mishautkin committed:
Update mollie-hosted-checkout pay method

- rename to payForOrder
- exclude url assertion from the method
- exclude return of orderId
- update checkout scenarios accordingly

mishautkin committed:
Add order and subscription notes assertions

mishautkin committed:
Add clearHtmlTags to assert order notes methods

mishautkin committed:
Fix subscription renewal scenario

mishautkin committed:
Add assertion prefix to order notes and subscription assertions

mishautkin committed:
Clean up plugin-foundation spec

mishautkin committed:
Fix refunds

mishautkin committed:
Add refund project to playwright.config

mishautkin committed:
Lint fix

mishautkin committed:
Merge pull request #1156 from mollie/dev/qa/add-subscriptions-refunds-tests

[QA] Add new subscriptions and refunds tests
